### PR TITLE
[auth-swift] Consistent structure for FirebaseAuthInterop

### PR DESCRIPTION
**Pull request by paulb777**
Follow up to #12379 with more integration fixes:

- FirebaseAuthInterop had the wrong path structure for CocoaPods/SPM compatibility
- FirebaseAuth-Swift.h depends on UIKit
- Interop had been implemented for dependent interops like AppCheck, but not Auth itself - the Storage integration tests failed to auth
- User.uid was incorrectly implemented as an optional - Database quickstart caught.
- Regression test added to Client test for the build failure caught by react-native tests
- TODO: add more regression tests for more of Firebase.h

Note: The functions quickstart will fail until FirebaseUI updates its minimum iOS version to 13+

#no-changelog

### CODE REVIEW

File: .github/workflows/storage.yml [hunk 1]
```diff
@@ -134,8 +134,8 @@ jobs:
           quickstart-ios/storage/GoogleService-Info.plist "$plist_secret"
     - name: Xcode
       run: sudo xcode-select -s /Applications/${{ matrix.xcode }}.app/Contents/Developer
-    - name: Test objc quickstart
-      run: ([ -z $plist_secret ] || scripts/third_party/travis/retry.sh scripts/test_quickstart.sh Storage true)
+    # - name: Test objc quickstart
+    #   run: ([ -z $plist_secret ] || scripts/third_party/travis/retry.sh scripts/test_quickstart.sh Storage true)
```
Comment (morganchen12) [inline .github/workflows/storage.yml:138]: Can this be removed entirely?

**Review by morganchen12 — APPROVED** [review]

**Comment by morganchen12** [comment]
Looks like the availability changes are causing functions tests to fail.

File: .github/workflows/storage.yml (hunk 1 above)
Comment (paulb777) [reply .github/workflows/storage.yml:138]: This is hopefully a short-term GHA flake.  There is a better comment around this in `main` that will get merged at the next merge.

**Review by paulb777 — COMMENTED** [review]
Thanks for the review!

**Comment by paulb777** [comment]
Yes. the Functions Quickstart will fail until FirebaseUI moves to iOS 13+

**Review by andrewheard — APPROVED** [review]

**Pull request merged**